### DISCUSSION TITLE
Is it possible to query across multiple partitions?

### DISCUSSION THREAD

**Discussion by marcelomachado** [apache/couchdb#3444] [Q&A]
I would like to query documents in multiple partitions. So, instead to use a string I would pass an array as a parameter.

I didn't find this ideia in the documentation, so I believe this might be a misunderstanding on my part. So my question is, why doesn't that make sense as a desirable feature?

**Comment by florianb** [comment]
Hi @marcelomachado,

i hadn't that use case yet, but as far as i understand the docs at https://docs.couchdb.org/en/stable/api/partitioned-dbs.html#db-partition-partition-all-docs:

> This endpoint is a convenience endpoint for automatically setting bounds on the provided partition range. Similar results can be had by using the global `/db/_all_docs` endpoint with appropriately configured values for `start_key` and `end_key`.

Just using the usual endpoints should allow you to query documents ignoring partition borders.

With other words, as long as you don't reduce the subset of queried documents either using the `/{db}/partition` endpoint or `start_key`/`end_key` you should be able to query all documents.

**Comment by marcelomachado** [comment]
Hi @florianb  thank you for the answer.

I will try to be more clear to see if we are aligned.

Using partitions is desirable for my application mainly considering scalability. I use partition to scope reduction, so we do not need to evaluate all documents. However, in some cases I need to get documents that are in different partitions and combine them. To do so, I currently need to make a query for each of the partitions. It is not interesting for me to query using the endpoint related to all_docs because in this case I lose the scope reduction feature. This implementation requires a number of I/O operations equivalent to the number of queries required to access each partition -- It is not good for performance. 

At a glance, considering the following dataset, my wish is to query all documents from `partition1` and `partition2` that have the `type` equal to `links` using only one I/O operation in the database (the answer must be docX and docY).

```
{
    "id": "partition1:docX",
    "type": "link"
}, 
{ 
    "id": "partition2:docY",
    "type": "link"
},
...
{
    ...
}
```

**Reply by florianb** [reply]
You're welcome and thank you for your clarification - i guess i should have been more clear on my answer.

Since you're able to use each available API endpoint to access documents across partition borders you might use the "default" query options (API endpoints **not** prefixed by `/db/_partition`), f.e. `/db/_find` (and `/db/_index`) to search your raw documents or  you might use design documents (`/db/_design/...`) to prepare and search an aggregated subset of the necessary information.

As long as you don't use the any of the "partition" endpoints the partition shouldn't be recognized . That saying, you **are**, of course, able to use the  partition endpoints (as well), in order to query your prepared indices and design documents actually respecting your partitions. But you don't have to use them just because you have a partition in place.

I hope this leads you into the right direction.
